service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerTest.java

```diff
@@ -216,6 +263,92 @@ void aboutToSubmitShouldPopulateHiddenCCDFieldsInStandardDirectionOrderToPersist
+                        .build())
+                    .build()))
+                .hearingDetails(ImmutableList.of(
+                    Element.<HearingBooking>builder()
+                        .id(UUID.randomUUID())
+                        .value(HearingBooking.builder().date(LocalDate.of(2020,10, 27)).build())
+                        .build())).build());
+
+        given(localAuthorityNameLookupConfiguration.getLocalAuthorityName(Mockito.any()))
+            .willReturn("Example Local Authority");
+
+        given(cafcassLookupConfiguration.getCafcass(LOCAL_AUTHORITY_CODE))
+            .willReturn(new CafcassLookupConfiguration.Cafcass(CAFCASS_NAME, COURT_EMAIL_ADDRESS));
+
+        given(hearingBookingService.getMostUrgentHearingBooking(Mockito.any())).willReturn(HearingBooking.builder()
+            .date(LocalDate.of(2020,10,27)).build());
+
+        given(dateFormatterService.formatLocalDateToString(Mockito.any(),Mockito.any()))
+            .willReturn("27 October 2020");
+    }
+
+    @Test
+    void shouldNotTriggerSDOEventWhenDraft() throws Exception {
+        Order order = Order.builder()
+            .orderStatus(OrderStatus.DRAFT)
+            .build();
+
+        CallbackRequest request = CallbackRequest.builder().caseDetails(CaseDetails.builder()
+            .data(ImmutableMap.<String, Object>builder()
+                .put("standardDirectionOrder", order)
+                .build()).build())
+            .build();
+
+        makeRequest(request, "submitted");
+
+        verify(applicationEventPublisher, Mockito.times(0)).publishEvent(Mockito.any());
+    }
+
+    @Test
+    void shouldTriggerSDOEventWhenSubmitted() throws Exception {
+        Order order = Order.builder()
+            .orderStatus(OrderStatus.SEALED)
+            .build();
+
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(12345L)
+                .data(ImmutableMap.<String, Object>builder()
+                    .put("hearingDetails",
+                        ImmutableList.of(
+                            Element.builder()
+                                .value(HearingBooking.builder()
+                                    .date(LocalDate.of(2020,10,20))
+                                    .build())
+                                .build()))
+                    .put("standardDirectionOrder", order)
+                    .put("caseLocalAuthority","example")
+                    .build())
+                .build())
+            .build();
+
+        final Map<String, Object> expectedCafcassParameters = ImmutableMap.<String, Object>builder()
+            .put("title","cafcass")
+            .put("familyManCaseNumber", "")
+            .put("leadRespondentsName", "")
+            .put("hearingDate","20 October 2020")
+            .put("reference", "12345")
+            .put("caseUrl", "http://fake-url/case/" + JURISDICTION + "/" + CASE_TYPE + "/12345")
+            .build();
```

Comment:
I would maybe extract to a private method, don't actually care what the contents are in this test